src/org/mozilla/javascript/NativeObject.java

```diff
@@ -472,6 +473,24 @@ public Object execIdCall(
                     }
                     return cx.newArray(scope, ids);
                 }
+            case ConstructorId_hasOwn:
+                {
+                    Object arg = args.length < 1 ? Undefined.instance : args[0];
+                    ScriptableObject obj = ensureScriptableObject(arg);
+                    Object str = args.length < 2 ? Undefined.instance : args[1];
+                    boolean result;
```

Comment:
I read the spec.
https://tc39.es/proposal-accessible-object-hasownproperty/

`Object.hasOwn` calls the `HasOwnProperty` operation in the same way as `Object.prototype.hasOwnProperty`. Could you make this common? I think it is better to write it in `ScriptRuntime` or `ScriptableObject`.
https://github.com/mozilla/rhino/blob/a0a9fb97f0439a978581283277325e382557ac67/src/org/mozilla/javascript/NativeObject.java#L207-L219

Reply:
Good point!

The correct location to put it however is org.mozilla.javascript.AbstractEcmaObjectOperations, as the spec defines the HasOwnProperty operation as one of the Abstract Operations on Objects (section 7.3.13).

Do cehck the notes at the top of the file as to how to add stuff: https://github.com/mozilla/rhino/blob/master/src/org/mozilla/javascript/AbstractEcmaObjectOperations.java#L8

Reply:
I changed 'hasOwnProperty' and 'hasOwn' to call the function 'hasOwnPropertyInObject' with the https://github.com/mozilla/rhino/pull/1157/commits/ec9b15047077626b59df3fc98407345f2787d433 commit. I'll have to find more codes to handle the same process here.